exchanges/gateio/gateio_websocket.go

```diff
@@ -816,3 +812,72 @@ func (g *Gateio) handleSubscription(ctx context.Context, conn stream.Connection,
 	}
 	return errs
 }
+
+// ResultHolder is used to unmarshal the result of a websocket request back to the required caller type
+type ResultHolder struct {
+	Result any `json:"result"`
+}
+
+// SendWebsocketRequest sends a websocket request to the exchange
+func (g *Gateio) SendWebsocketRequest(ctx context.Context, epl request.EndpointLimit, channel string, connSignature, params, result any, expectedResponses int) error {
+	paramPayload, err := json.Marshal(params)
+	if err != nil {
+		return err
+	}
+
+	conn, err := g.Websocket.GetConnection(connSignature)
+	if err != nil {
+		return err
+	}
+
+	tn := time.Now().Unix()
+	req := &WebsocketRequest{
+		Time:    tn,
+		Channel: channel,
+		Event:   "api",
+		Payload: WebsocketPayload{
+			// This request ID associated with the payload is the match to the
+			// response.
+			RequestID:    strconv.FormatInt(conn.GenerateMessageID(false), 10),
+			RequestParam: paramPayload,
+			Timestamp:    strconv.FormatInt(tn, 10),
+		},
+	}
+
+	responses, err := conn.SendMessageReturnResponsesWithInspector(ctx, epl, req.Payload.RequestID, req, expectedResponses, wsRespAckInspector{})
+	if err != nil {
+		return err
+	}
+
+	if len(responses) == 0 {
+		return common.ErrNoResponse
+	}
+
+	var inbound WebsocketAPIResponse
+	// The last response is the one we want to unmarshal, the other is just
+	// an ack. If the request fails on the ACK then we can unmarshal the error
+	// from that as the next response won't come anyway.
+	endResponse := responses[len(responses)-1]
```

Comment:
Think we can be clearer about this.
```suggestion
	// responses may include an ack resp, which we skip
	resp := responses[len(responses)-1]
```
However, might we want to assert len(responses) <= 2 ?

Reply:
Sounds good 👍